src/MiniCover/HitServices/HitServiceWithTests.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace MiniCover
+{
+    public static class HitServiceWithTests
+    {
+        private static readonly object lockObject = new object();
+        private static Dictionary<string, Dictionary<int, HitTrace>> files = new Dictionary<string, Dictionary<int, HitTrace>>();
+        private static Dictionary<string, bool> assemblyPdb = new Dictionary<string, bool>();
+        public static void Init(string fileName)
+        {
+            lock (lockObject)
+            {
+                if (!files.ContainsKey(fileName))
+                {
+                    files[fileName] = new Dictionary<int, HitTrace>();
+                }
+            }
+        }
+
+        public static void Hit(string fileName, int id)
+        {
+            lock (lockObject)
+            {
+                StackTrace stackTrace = new StackTrace();
+
+                var testMethod = GetTestMethod(stackTrace);
+                
+                var hits = files[fileName];
+                if (!hits.ContainsKey(id))
+                    hits.Add(id, new HitTrace(id));
+                hits[id].Hited(testMethod);
+            }
+        }
+
+        private static MethodBase GetTestMethod(StackTrace stack)
+        {
+            var frames = stack.GetFrames();
+            for (int i = frames.Length - 1; i >= 0; i--)
+            {
+                var currentMethod = frames[i].GetMethod();
+                if (currentMethod.HasPdb())
+                    return currentMethod;
+            }
+
+            return null;
+        }
+
+        private static bool HasPdb(this MethodBase methodBase)
+        {
+            var location = methodBase.DeclaringType.Assembly.Location;
+
+            if (!assemblyPdb.TryGetValue(location, out var hasPdb))
+            {
+                assemblyPdb[location] = hasPdb = File.Exists(Path.ChangeExtension(location, ".pdb"));
+            }
+            return hasPdb;
+        }
+
+        static void Save()
+        {
+            foreach (var fileName in files)
+            {
+                using (var streamWriter = new StreamWriter(File.Open(fileName.Key, FileMode.Append)))
+                {
+                    foreach (var hitTrace in fileName.Value)
+                    {
+                        hitTrace.Value.WriteInformation(streamWriter);
+                    }
+
+                    streamWriter.Flush();
+                }
+            }
+        }
+
+        static HitServiceWithTests()
+        {
+            AppDomain.CurrentDomain.ProcessExit += (sender, args) => Save();
```

Comment:
I was running some tests and noticed some hits was missing.
Then I learned `ProcessExit` only have 2 seconds to run, and the writing of hits is interrupted.
https://msdn.microsoft.com/en-us/library/system.appdomain.processexit%28v=vs.110%29.aspx?f=255&MSPPError=-2147217396
Do you know how we could workaround that issue?

Reply:
I don't for now but going to find

Reply:
Maybe that's not an issue, if we make hits file more lightweight.
I'm reviewing and trying to optimize a bit further. I will create a PR to your branch.